gapis/trace/android/mali/profiling_data.go

```diff
@@ -252,6 +253,13 @@ func processGpuSlices(ctx context.Context, processor *perfetto.Processor, captur
 }
 
 func processCounters(ctx context.Context, processor *perfetto.Processor, desc *device.GpuCounterDescriptor) ([]*service.ProfilingData_Counter, error) {
+	// Use counter name, rather than counter id as key here, since the counter id
+	// is different from the counter track id that we would query.
+	specMapping := map[string]*device.GpuCounterDescriptor_GpuCounterSpec{}
+	for _, spec := range desc.GetSpecs() {
+		specMapping[spec.GetName()] = spec
+	}
```

Comment:
Could this mapping be built once-for-all rather than every time processCounters() is called?

It looks like this mapping only depends on `desc`, which itself only depends on the device configuration, see how `desc` is obtained in gapis/trace/android/trace.go:ProcessProfilingData(). If so, we could create this mapping once and somehow attach it to the device binding information, to avoid re-creating it everytime here. What do you think?